src/polyswarm/client/search.py

```diff
@@ -55,13 +55,23 @@ def urls(ctx, url):
               help='Field to be included in the result (.* wildcards are accepted).')
 @click.option('-x', '--exclude', type=click.STRING, multiple=True,
               help='Field to be excluded from the result (.* wildcards are accepted).')
-@click.argument('query_string', nargs=-1, required=True)
+@click.option('-p', '--ip', type=click.STRING, multiple=True,
+              help='IP address IOC to search')
+@click.option('-u', '--url', type=click.STRING, multiple=True,
+              help='URL address IOC to search')
+@click.option('-d', '--domain', type=click.STRING, multiple=True,
+              help='Domain address IOC to search')
+@click.argument('query_string', nargs=-1, required=False)
 @click.pass_context
-def metadata(ctx, query_string, include, exclude):
+def metadata(ctx, query_string, include, exclude, ip, url, domain):
     api = ctx.obj['api']
     output = ctx.obj['output']
     query_string = ' '.join(query_string)
-    for metadata_result in api.search_by_metadata(query_string, include=include, exclude=exclude):
+
+    if sum([not ip, not url, not domain]) < 2:
+        raise click.exceptions.BadParameter("Only one of the following paramters may be provided per request: ip, url, or domain")
```

Comment:
@sbneto @mjbradford89 do we want to move this mutually exclusive check to be in the API? And then have a try/except around the for loop below here?

Reply:
After adding the check in artifact-index and removing the one in the cli, the error the user sees is:

```bash
error [polyswarm.client.polyswarm]: Return code: 400
error [polyswarm.client.polyswarm]: Message: The options 'ip', 'domain', and 'url' are mutually exclusive.
```

IMO this may now be good enough but I can also add some custom logic here in the cli to print a cleaner error if necessary.

Reply:
that works for me.